### DISCUSSION TITLE
[@types/supertest] Cannot find name 'Request'

### DISCUSSION THREAD

**Discussion by jcvidiri** [DefinitelyTyped/DefinitelyTyped#68714] [Issues with a @types package] [answered]
When attempting to build a typescript project with `@types/supertest` installed, i get this error

```ts
node_modules/@types/superagent/lib/node/response.d.ts:13:26 - error TS2304: Cannot find name 'Request'.

13     constructor(request: Request);
                            ~~~~~~~

node_modules/@types/superagent/lib/node/response.d.ts:34:34 - error TS2304: Cannot find name 'Request'.

34     request: InstanceType<typeof Request>;
                                    ~~~~~~~
```

I found the issue with version `2.0.10` but also continue to experience it when updating to latest (`6.0.2`).
This project uses node `18.18.0`, but updating to version `20.8.1` shows the same

other relevant versions:
```
"typescript": "4.6.3"
"@types/node": "18.18.0",
```

[a comment by an automated account is not shown]

**Reply by jcvidiri** [reply]
my bad! fixed it, the issue is related to `@types/supertest`

**Comment by DavidTanner** [comment, marked as answer]
I have a PR open to fix the issue, just waiting on reviews.

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68402

**Reply by jcvidiri** [reply]
Oh nice! thanks 👍
